targets/psa_switch/externs/psa_meter.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2020-present Cornell University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Yunhe Liu (yunheliu@cs.cornell.edu)
+ *
+ */
+
+#ifndef PSA_SWITCH_PSA_METER_H_
+#define PSA_SWITCH_PSA_METER_H_
+
+#include <bm/bm_sim/extern.h>
+#include <bm/bm_sim/meters.h>
+
+namespace bm {
+
+namespace psa {
+
+class PSA_Meter : public bm::ExternType {
+ public:
+  static constexpr p4object_id_t spec_id = 0xfffffffe;
+
+  BM_EXTERN_ATTRIBUTES {
+    BM_EXTERN_ATTRIBUTE_ADD(n_meters);
+    BM_EXTERN_ATTRIBUTE_ADD(type);
+    BM_EXTERN_ATTRIBUTE_ADD(is_direct);
+    BM_EXTERN_ATTRIBUTE_ADD(rate_count);
+  }
+  
+  void init() override {
```

Comment:
looks like this definition could go into the .cpp file

Reply:
Thanks for the comments! All make sense, working on the changes! 

> > Has not implemented psa-meter RuntimeCmd yet.
> 
> You only need to implement the runtime interface I showed you last time, which should be a couple lines of code, in order for this to work. Then you will be able to write an STF test in the compiler.

For runtime interface, please correct me if I think wrong, I currently see `meter_array_set_rates`, `meter_set_rates` and `meter_get_rates` in [switch.h](https://github.com/p4lang/behavioral-model/blob/master/include/bm/bm_sim/switch.h#L718), I will add corresponding runtime interfaces into [psa_switch.h](https://github.com/p4lang/behavioral-model/blob/master/targets/psa_switch/psa_switch.h#L114).

Reply:
correct

Reply:
@antoninbas 
- RuntimeCmd Interfaces added (`meter_array_set_rates`, `meter_set_rates`, `meter_get_rates`). 
- Other change request updated as well. 

Would you please take a look and let me know if any further change is needed? Otherwise, I think this PR is ready to be merged? Thanks much!